lib/flipper/test_help.rb

```diff
@@ -0,0 +1,36 @@
+module Flipper
+  module TestHelp
+    def flipper_configure
+      # Create a single shared memory adapter instance for each test
+      @flipper_adapter = Flipper::Adapters::Memory.new
+
+      Flipper.configure do |config|
+        config.adapter { @flipper_adapter }
+        config.default { Flipper.new(config.adapter) }
+      end
+    end
+
+    def flipper_reset
+      Flipper.instance = nil # Reset previous flipper instance
```

Comment:
I got some spec failures after this - you get a new flipper instance but using the same memory adapter, so there's  leakage between tests. 

I can make everything green again if I add `Flipper.instance.import(Flipper::Adapters::Memory.new)` to the reset (which was the simplest way I could find to delete all the data from the shared adapter)

Reply:
I think I hit the same. Maybe it would be better to just disable all feature flags in `reset`?

Reply:
Yes, I hit this. The app thread and spec thread end up using a different flipper instance (fine). The app thread is running for the entire duration of the specs so keeps the same flipper instance.

On the second spec the before each creates a new flipper instance but this only affects the spec thread, the app thread still has the old memory adapter.

To make this work I had to disable the builtin flipper/test_help (with the env var) and then do what I was doing before: create the shared memory adapter once in before(suite) and clear out its contents in before each